From da4c797e8f99253f371d07478ccbc2abb0ed1a7e Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Mon, 18 May 2020 21:32:22 +0200
Subject: [PATCH] PCI: sanity test on PCI vendor to be sure we do not touch
 everything
Patch-mainline: Never, Suse work around
References: bsc#1141558

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/pci/hotplug/pciehp_core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/pci/hotplug/pciehp_core.c b/drivers/pci/hotplug/pciehp_core.c
index 6782e9e..e541730 100644
--- a/drivers/pci/hotplug/pciehp_core.c
+++ b/drivers/pci/hotplug/pciehp_core.c
@@ -214,6 +214,9 @@ static void check_for_broken_active_bit(struct controller *ctrl)
 	if (!product_name)
 		return;
 
+	if (ctrl->pcie->port->vendor != 0x10b5 && ctrl->pcie->port->vendor != 0x8086)
+		return;
+
 	if (	strstr(product_name, "PRIMEQUEST 2400E3") ||
 		strstr(product_name, "PRIMEQUEST 2400L3") ||
 		strstr(product_name, "PRIMEQUEST 2800E3") ||
-- 
2.16.4

